### ISSUE TITLE
hasOwnProperty should be called via Object.prototype.hasOwnProperty

### ISSUE THREAD

**Issue by ibukanov** [wycats/handlebars.js#386]
Handlebars each implementation https://github.com/wycats/handlebars.js/blob/master/lib/handlebars/base.js#L80 contains:

```
  for(var key in context) {
    if(context.hasOwnProperty(key)) {
```

Here hasOwnProperty is called on an object that may have arbitrary properties including hasOwnProperty. To protect against that Object.prototype.hasOwnProperty.call(context, key) should be used instead.

**Comment by julien-f** [comment]
It is [common practice](http://www.2ality.com/2013/10/dict-pattern.html) to create an object to use as a dictionary with a `null` prototype.

Unfortunately they cannot be used with Handlebars because of the use of `hasOwnProperty()` as a method.

**Comment by cdhowie** [comment]
@wycats Please consider reopening this.  Consider, for example, passing data parsed as a query string to a Handlebars template, then using `{{each}}` on that object.  If the query string contains `hasOwnProperty=foo` then this will cause an exception for no good reason.  It seems like a bad idea to allow user-supplied input to break a template.

The alternative is that I have to enumerate the properties of `Object.prototype` and specifically delete each property from the data object.

**Comment by nknapp** [comment]
I am re-opening this because there is even an eslint-rule against this (https://eslint.org/docs/rules/no-prototype-builtins) 

Generally, I think we should go from our custom eslint-rules to use the recommended rule-set.
And use prettier for formatting instead of eslint. (like in https://github.com/handlebars-lang/parser, but without typescript.) That would show such deficits better.

**Comment by nknapp** [comment]
Should have been solved in f7f05d7558e674856686b62a00cde5758f3b7a08
